test/FunctionalTests/Microsoft.OData.Core.Tests/ScenarioTests/UriParser/EnumFilterFunctionalTests.cs

```diff
@@ -515,13 +520,205 @@ public void ParseFilterEnumTypesWrongCast3()
+            leftNode.ShouldBeEnumNode(colorType, enumMember.Name);
+
+            Assert.True(leftNode.TypeReference.IsEnum());
+            Assert.Equal(enumMember.Name, leftNode.Value.ToString());
+            Assert.Equal(expectedLiteral, leftNode.LiteralText);
+
+            CollectionPropertyAccessNode rightNode = Assert.IsType<CollectionPropertyAccessNode>(inNode.Right);
+            rightNode.ShouldBeCollectionPropertyAccessQueryNode(this.GetColorsProperty(this.userModel));
+            Assert.Equal(colorType, rightNode.ItemType.Definition);
+        }
+
+        [Fact]
+        public void ParseFilterWithInOperatorWithEnumsMemberDoubleIntegralValue()
+        {
+            // Arrange
+            long enumValue = 3;
+            string filterQuery = $"{enumValue} in Colors"; // "3 in Colors"
+            string expectedLiteral = "'Blue'";
+
+            IEdmEnumType colorType = this.GetColorType(this.userModel);
+            bool success = colorType.TryParse(enumValue, out IEdmEnumMember enumMember);
+
+            // Act
+            FilterClause filterQueryNode = ParseFilter(filterQuery, this.userModel, this.entityType, this.entitySet);
+            SingleValueNode expression = filterQueryNode.Expression;
+
+            // Assert
+            Assert.Equal("3 in Colors", filterQuery);
+            Assert.True(success);
+            InNode inNode = expression.ShouldBeInNode();
+            ConstantNode leftNode = Assert.IsType<ConstantNode>(inNode.Left);
+            leftNode.ShouldBeEnumNode(colorType, enumMember.Name);
+
+            Assert.True(leftNode.TypeReference.IsEnum());
+            Assert.Equal(enumMember.Name, leftNode.Value.ToString());
+            Assert.Equal(expectedLiteral, leftNode.LiteralText);
+
+            CollectionPropertyAccessNode rightNode = Assert.IsType<CollectionPropertyAccessNode>(inNode.Right);
+            rightNode.ShouldBeCollectionPropertyAccessQueryNode(this.GetColorsProperty(this.userModel));
+            Assert.Equal(colorType, rightNode.ItemType.Definition);
+        }
+
+        [Theory]
+        [InlineData("3.0")]
+        [InlineData("3.1")]
+        [InlineData("4.0")]
+        [InlineData("5.0")]
+        public void ParseFilterWithInOperatorWithEnumsMemberFloatIntegralValue(string floatString)
+        {
+            // Arrange
+            string filterQuery = $"{floatString} in Colors";
+
+            // Act
+            Action test = () => ParseFilter(filterQuery, this.userModel, this.entityType, this.entitySet);
+
+            // Assert
+            test.Throws<ArgumentException>(Strings.Nodes_InNode_CollectionItemTypeMustBeSameAsSingleItemType("NS.Color", "Edm.Single")); // Float are of Type Edm.Single
+        }
+
+        [Theory]
+        [InlineData(-20)]
+        [InlineData("-20")]
+        [InlineData("'-20'")]
+        public void ParseFilterWithInOperatorWithEnumsMemberInvalidIntegralValue(object integralValue)

```

Comment:
Add a test to validate that an error is thrown if we pass a string that doesn't exist
E.g `$filter='Blue10' in Colors`

Reply:
Thanks @KenitoInc for this. I have added the tests. Please see using the link below:

- [ParseFilterWithInOperatorWithEnumsInvalidMemberNames_ThrowsIsNotValidEnumConstantException](https://github.com/OData/odata.net/pull/3023/files#diff-9f67b5b315ba1bb4ddbcb823f5c4183658c6a7f7a3b06d0da900b0ea7e21c7d6:~:text=ParseFilterWithInOperatorWithEnumsInvalidMemberNames_ThrowsIsNotValidEnumConstantException)